src/server/models/Reading.js

```diff
@@ -143,6 +150,32 @@ class Reading {
 		return compressedReadingsByMeterID;
 	}
 
+	/**
+	 * Gets a number of aggregate readings across the given time interval for the given meters.
+	 *
+	 * Compressed readings are in kilowatts.
+	 * @param meterIDs an array of ids for meters whose points are being compressed
+	 * @param duration A moment time duration over which to sum the readings
+	 * @param fromTimestamp An optional start point for the beginning of the entire time range.
+	 * @param toTimestamp An optional end point for the end of the entire time range.
+	 * @param conn the connection to use. Defaults to the default database connection.
+	 * @return {Promise<object<int, array<{reading_sum: number, start_timestamp: Date, end_timestamp: Date}>>>}
+	 */
+	static async getAggregateReadings(meterIDs, duration, fromTimestamp = null, toTimestamp = null, conn = db) {
```

Comment:
I don't think this is named well. It doesn't convey the fact that the readings are for the bar chart.